File: backend/app/schemas/consumer.py
from datetime import datetime

from pydantic import BaseModel


class PipelineConsumerSchema(BaseModel):
    pipeline_id: str
    pipeline_name: str
    dag_id: str
    airflow_status: str  # success, failed, running, unknown
    last_run_at: datetime | None = None


class PipelineConsumersResponse(BaseModel):
    consumers: list[PipelineConsumerSchema]

File: frontend/src/types/consumer.ts
export interface PipelineConsumer {
  pipeline_id: string;
  pipeline_name: string;
  dag_id: string;
  airflow_status: string;
  last_run_at: string | null;
}

export interface PipelineConsumersResponse {
  consumers: PipelineConsumer[];
}
